src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -64,17 +64,28 @@ public override bool Execute()
                 Log.LogError("Cannot find specified project.json - '{0}'", ProjectJson);
                 return false;
             }
-            if (!string.IsNullOrWhiteSpace(PackageBuildNumberOverride))
+
+            List<ITaskItem> packageInformation = new List<ITaskItem>();
```

Comment:
A dictionary would be nice for lookup purposes, but we have multiple possible input sources (various package drops, or versions files), and the code is not currently written to guarantee a unique key.  I'll file an issue to investigate refactoring this, but I'll leave this as a list for now.